### PR TITLE
chore(common): implement enforced pre-commit type checks for FE service files

### PR DESCRIPTION
**Pull request by jamesgeorge007**
### Description

This PR aims at performing type checks for TS files for FE services source files as part of the pre-commit hook.

- Adds a `type-check.mjs` script at the `hoppscotch-common` package root that performs type check on TS files under the `services` directory leveraging the TS and Vue TSC compiler APIs. Files under `**/__tests__/**` & `.d.ts` based are ignored while performing type check. The `findFilesToPerformTypeCheck` helper function accepts the list of directory paths and file patterns where type checking will be performed.
- Adds `glob` as a dev dependency consumed while compiling TS files in the given directory.
- Update the `do-typecheck` script to invoke the above script, configured to run as part of the pre-commit hook.
- Fixes an existing type error in scope.

Closes HFE-249.

### Demo

https://github.com/hoppscotch/hoppscotch/assets/25279263/1b356afb-d6e7-4721-875d-4aa50b933ea6

### Checks
- [x] My pull request adheres to the code style of this project
- [x] All the tests have passed

### CODE REVIEW

File: packages/hoppscotch-common/package.json [hunk 1]
```diff
@@ -136,6 +136,7 @@
     "eslint": "^8.47.0",
     "eslint-plugin-prettier": "^5.0.0",
     "eslint-plugin-vue": "^9.17.0",
+    "glob": "^10.3.10",
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/package.json:139]: Isn't glob a dev-dependency in this case ?

**Review by AndrewBastin — CHANGES_REQUESTED** [review]

File: packages/hoppscotch-common/package.json (hunk 1 above)
Comment (jamesgeorge007) [reply packages/hoppscotch-common/package.json:139]: It's already present under `devDependencies`.

File: packages/hoppscotch-common/type-check.mjs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const fileList = files
+    // Narrow down to `TS` files
+    .filter((file) => file.endsWith(".ts"))
+    // Append the path
+    .map((file) => path.join(directoryPath, file))
+
+  return fileList
+}
+
+const directoryPath = path.resolve(__dirname, "src", "services")
```
[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/hoppscotch-common/type-check.mjs]: Updated the `findTypeScriptFiles` function to accept the list of directory paths where it'll scan for TS files. Moved the helper functions to the top of the file.

File: packages/hoppscotch-common/package.json (hunk 1 above)
Comment (AndrewBastin) [reply packages/hoppscotch-common/package.json:139]: oops lol. missed that due to the fold.

**Review by AndrewBastin — APPROVED** [review]

**Pull request merged**